plugins/woocommerce/includes/class-wc-order.php

```diff
@@ -103,25 +103,25 @@ public function payment_complete( $transaction_id = '' ) {
 		}
 
 		try {
-			do_action( 'woocommerce_pre_payment_complete', $this->get_id() );
+			do_action( 'woocommerce_pre_payment_complete', $this->get_id(), $transaction_id );
 
 			if ( WC()->session ) {
 				WC()->session->set( 'order_awaiting_payment', false );
 			}
 
-			if ( $this->has_status( apply_filters( 'woocommerce_valid_order_statuses_for_payment_complete', array( 'on-hold', 'pending', 'failed', 'cancelled' ), $this ) ) ) {
+			if ( $this->has_status( apply_filters( 'woocommerce_valid_order_statuses_for_payment_complete', array( 'on-hold', 'pending', 'failed', 'cancelled' ), $this, $transaction_id ) ) ) {
```

Comment:
Adding `$transaction_id` isn't particularly problematic, but I wonder if we should really be adding it. Is this needed for your purposes or would it be enough with `$transaction_id` being passed to the action hooks (not filters)?